### PR TITLE
fix(shared_sub): Run mnesia operations in a middleman process

### PR DESCRIPTION
**Pull request by ieQu1**

### CODE REVIEW

File: src/emqx_misc.erl [hunk 1]
```diff
@@ -330,6 +331,39 @@ hexchar2int(I) when I >= $0 andalso I =< $9 -> I - $0;
...
+                                        {EC, Err}
+                                end,
+                       Parent ! {Ref, Result}
+               end),
+    receive
+        {Ref, Result} ->
+            case Result of
+                {ok, R} -> R;
+                {EC, Err} ->
+                    erlang:EC(Err)
```
Comment (zmstone) [inline src/emqx_misc.erl:363]: Raise with original stacktrace?

Comment (ieQu1) [reply src/emqx_misc.erl:363]: How do I do that?

Comment (thalesmg) [reply src/emqx_misc.erl:363]: Maybe [`erlang:raise/3`](https://www.erlang.org/doc/man/erlang.html#raise-3) can do that.

Comment (ieQu1) [reply src/emqx_misc.erl:363]: Thanks, I didn't know about this function. However, mnesia only exits, which doesn't produce a stacktrace under normal conditions (unless you catch). So in this case it's probably unnecessary.

Comment (zmstone) [reply src/emqx_misc.erl:363]: yes, catch and return at line 354, and call `erlang:raise/3` here

```
                                    EC:Err:Stack ->
                                        {EC, Err, Stack}
```

Comment (zmstone) [reply src/emqx_misc.erl:363]: ah. if the stack is not interesting anyway. then ignore this.

**Review by zmstone — APPROVED** [review]

**Comment by zmstone** [comment]
should be sent to `main-v4.3` ?

File: scripts/update-appup.sh [hunk 2]
```diff
@@ -123,8 +123,12 @@ PREV_REL_DIR="${PREV_DIR_BASE}/${PREV_TAG}/_build/${PROFILE}/lib"
 if [ "${IS_CHECK:-}" = 'yes' ]; then
     diffs="$(git diff --name-only | grep -E '\.appup\.src' || true)"
     if [ "$diffs" != '' ]; then
-        git --no-pager diff
-        echo "$0 ---check produced git diff"
+        git add "$diffs"
+        git commit -m "chore: Update appup"
```
Comment (thalesmg) [inline scripts/update-appup.sh:127]: Maybe we could only print the warning message, but let the dev inspect and stage the changes manually?

Comment (ieQu1) [reply scripts/update-appup.sh:127]: This is not going to be committed to the main repo, it's just a quick & dirty way to format a patch.

**Comment by ieQu1** [comment]
Merged from 4.3

**Pull request closed without merging**